modules/mobile-agents/windows/jax-rs/src/main/java/org/wso2/carbon/mdm/mobileservices/windows/services/devicemgtservice/DeviceManagementService.java

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright (c) 2015, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.mdm.mobileservices.windows.services.devicemgtservice;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.device.mgt.common.Device;
+import org.wso2.carbon.device.mgt.common.DeviceIdentifier;
+import org.wso2.carbon.device.mgt.common.DeviceManagementConstants;
+import org.wso2.carbon.device.mgt.common.DeviceManagementException;
+import org.wso2.carbon.device.mgt.common.license.mgt.License;
+import org.wso2.carbon.mdm.mobileservices.windows.common.exceptions.WindowsConfigurationException;
+import org.wso2.carbon.mdm.mobileservices.windows.common.util.Message;
+import org.wso2.carbon.mdm.mobileservices.windows.common.util.WindowsAPIUtils;
+
+import javax.jws.WebService;
+import javax.ws.rs.*;
+import javax.ws.rs.core.Response;
+import java.util.List;
+
+/**
+ * Windows Device Management REST-API implementation.
+ * All end points supports JSON, XMl with content negotiation.
+ */
+@WebService
+@Produces({"application/json", "application/xml"})
+@Consumes({"application/json", "application/xml"})
+public class DeviceManagementService {
+
+    private static Log log = LogFactory.getLog(DeviceManagementService.class);
+
+    /**
+     * Get all devices.Returns list of Windows devices registered in MDM.
+     *
+     * @return Device List
+     * @throws WindowsConfigurationException
```

Comment:
It is better if the return values and exception (reason for exception etc) are described well.